### PR TITLE
docs: add note about support regarding version upgrade

### LINKED ISSUE #552: Document supported / tested upgrade scenario
### Describe the Issue

Most of customers are running v21.11.x right now. We've released v22.1 and will release v22.2 soon, but we don't have a clear guideline on what's the supported upgrade path from v21.11.x to v22.2. 

### Updates to existing documentation

https://docs.redpanda.com/docs/cluster-administration/version-upgrade/ is missing the information

### New feature or content gap requests

The core engineering team is testing the upgrade scenario over https://github.com/redpanda-data/redpanda/issues/5920, therefore the doc should align with the result and give a proper guideline to the readers.

### Additional notes

Slack
#release-22_2_1 https://redpandadata.slack.com/archives/C03PGC3DWVD/p1660023232134759
#core https://redpandadata.slack.com/archives/C01ND4SVB6Z/p1660021226351389

### PR DESCRIPTION
**Pull request by Deflaimun**
Resolves #552 

Preview: 

https://deploy-preview-859--redpanda-documentation.netlify.app/docs/platform/install-upgrade/version-upgrade/#upgrading-your-version

https://deploy-preview-859--redpanda-documentation.netlify.app/docs/22.2/platform/install-upgrade/version-upgrade/#upgrading-your-version

https://deploy-preview-859--redpanda-documentation.netlify.app/docs/22.1/cluster-administration/version-upgrade/#upgrading-your-version

https://deploy-preview-859--redpanda-documentation.netlify.app/docs/21.11/cluster-administration/version-upgrade/#upgrading-your-version

### CODE REVIEW

File: docs/platform/install-upgrade/version-upgrade.mdx [hunk 1]
```diff
@@ -85,6 +85,10 @@ You can check which versions are available in the [release notes](https://github
 
 ## Upgrading your version
 
+:::note
+We recommend upgrading only one major version at a time. For example, from `22.2` to `22.3`.
```
Comment (Feediver1) [inline docs/platform/install-upgrade/version-upgrade.mdx:89]:
```suggestion
Redpanda recommends that you upgrade only one major version at a time. For example, from `22.2` to `22.3`.
```

File: versioned_docs/version-21.11/cluster-administration/version-upgrade.mdx [hunk 2]
```diff
@@ -27,6 +27,12 @@ The expected output will look similar to:
 v21.11.5 (rev af88fa1)
 ```
 
+## Upgrading your version
+
+:::note
+We recommend upgrading only one major version at a time. For example, from `22.2` to `22.3`.
```
Comment (Feediver1) [inline versioned_docs/version-21.11/cluster-administration/version-upgrade.mdx:33]:
```suggestion
Redpanda recommends that you upgrade only one major version at a time. For example, from `22.2` to `22.3`.
```

File: versioned_docs/version-22.1/cluster-administration/version-upgrade.mdx [hunk 3]
```diff
@@ -85,6 +85,10 @@ You can check which versions are available in the [release notes](https://github
 
 ## Upgrading your version
 
+:::note
+We recommend upgrading only one major version at a time. For example, from `22.2` to `22.3`.
```
Comment (Feediver1) [inline versioned_docs/version-22.1/cluster-administration/version-upgrade.mdx:89]:
```suggestion
Redpanda recommends that you upgrade only one major version at a time. For example, from `22.2` to `22.3`.
```

File: versioned_docs/version-22.2/platform/install-upgrade/version-upgrade.mdx [hunk 4]
```diff
@@ -85,6 +85,10 @@ You can check which versions are available in the [release notes](https://github
 
 ## Upgrading your version
 
+:::note
+We recommend upgrading only one major version at a time. For example, from `22.2` to `22.3`.
```
Comment (Feediver1) [inline versioned_docs/version-22.2/platform/install-upgrade/version-upgrade.mdx:89]:
```suggestion
Redpanda recommends that you upgrade only one major version at a time. For example, from `22.2` to `22.3`.
```

**Review by Feediver1 — APPROVED** [review]
Please implement edits. Also, @daisukebe and @mattschumpert must approve this one. Thx @Deflaimun !

File: docs/platform/install-upgrade/version-upgrade.mdx [hunk 5]
```diff
@@ -85,6 +85,10 @@ You can check which versions are available in the [release notes](https://github
 
 ## Upgrading your version
 
+:::note
+Redpanda recommends that you upgrade only one major version at a time. For example, from `22.2` to `22.3`.
```
Comment (jcsp) [inline docs/platform/install-upgrade/version-upgrade.mdx:89]: "recommend" is too soft of a word.  This is the _only_ supported upgrade path: it is mandatory.

For best results we also recommand updating to the latest minor release of your current feature branch before updating to the next feature branch.  This part isn't mandatory.

Comment (Deflaimun) [reply docs/platform/install-upgrade/version-upgrade.mdx:89]: How about 
"Redpanda supports upgrading only one major version at a time. For example, from `22.2` to `22.3`."
@jcsp

**Review by mattschumpert — APPROVED** [review]

File: docs/platform/install-upgrade/version-upgrade.mdx (hunk 5 above)
Comment (jcsp) [reply docs/platform/install-upgrade/version-upgrade.mdx:89]: I would phrase it something like this:

When upgrading between major versions, it is only supported upgrade one version at a time.  For example, upgrading from `22.1` to `22.2` is supported, while upgrading from `22.1` directly to `22.3` is not.  In the latter case, you would upgrade first to `22.2`, and then onwards to `22.3`

Comment (Feediver1) [reply docs/platform/install-upgrade/version-upgrade.mdx:89]: How about:
When upgrading between major versions, Redpanda only supports upgrading one version at a time. For example, upgrading from 22.1 to 22.2 is supported, while upgrading from 22.1 directly to 22.3 is not. In the latter case, you would upgrade first to 22.2, and then to 22.3

Comment (jcsp) [reply docs/platform/install-upgrade/version-upgrade.mdx:89]: :ok_hand: Perfect

Comment (Feediver1) [inline docs/platform/install-upgrade/version-upgrade.mdx:89]:
```suggestion
When upgrading between major versions, Redpanda only supports upgrading one version at a time. For example, upgrading from `22.1` to `22.2` is supported, while upgrading from `22.1` directly to `22.3` is not. In the latter case, you would upgrade first to `22.2`, and then to `22.3`.
```

File: versioned_docs/version-21.11/cluster-administration/version-upgrade.mdx [hunk 6]
```diff
@@ -27,6 +27,12 @@ The expected output will look similar to:
 v21.11.5 (rev af88fa1)
 ```
 
+## Upgrading your version
+
+:::note
+Redpanda recommends that you upgrade only one major version at a time. For example, from `22.2` to `22.3`.
```
Comment (Feediver1) [inline versioned_docs/version-21.11/cluster-administration/version-upgrade.mdx:33]:
```suggestion
When upgrading between major versions, Redpanda only supports upgrading one version at a time. For example, upgrading from `22.1` to `22.2` is supported, while upgrading from `22.1` directly to `22.3` is not. In the latter case, you would upgrade first to `22.2`, and then to `22.3`.
```

File: versioned_docs/version-22.1/cluster-administration/version-upgrade.mdx [hunk 7]
```diff
@@ -85,6 +85,10 @@ You can check which versions are available in the [release notes](https://github
 
 ## Upgrading your version
 
+:::note
+Redpanda recommends that you upgrade only one major version at a time. For example, from `22.2` to `22.3`.
```
Comment (Feediver1) [inline versioned_docs/version-22.1/cluster-administration/version-upgrade.mdx:89]:
```suggestion
When upgrading between major versions, Redpanda only supports upgrading one version at a time. For example, upgrading from `22.1` to `22.2` is supported, while upgrading from `22.1` directly to `22.3` is not. In the latter case, you would upgrade first to `22.2`, and then to `22.3`.
```

File: versioned_docs/version-22.2/platform/install-upgrade/version-upgrade.mdx [hunk 8]
```diff
@@ -85,6 +85,10 @@ You can check which versions are available in the [release notes](https://github
 
 ## Upgrading your version
 
+:::note
+Redpanda recommends that you upgrade only one major version at a time. For example, from `22.2` to `22.3`.
```
Comment (Feediver1) [inline versioned_docs/version-22.2/platform/install-upgrade/version-upgrade.mdx:89]:
```suggestion
When upgrading between major versions, Redpanda only supports upgrading one version at a time. For example, upgrading from `22.1` to `22.2` is supported, while upgrading from `22.1` directly to `22.3` is not. In the latter case, you would upgrade first to `22.2`, and then to `22.3`.
```

**Review by daisukebe — APPROVED** [review]
+1

**Pull request merged**